x/ccv/child/keeper/hooks.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+var _ slashingtypes.SlashingHooks = Keeper{}
+
+// AfterValidatorDowntime gets the given validator address jailing time
+// in order either to add it the downtime jailing duration and initiated its slashing
+// on the provider chain or perfom a no-op
+func (k Keeper) AfterValidatorDowntime(ctx sdk.Context, consAddr sdk.ConsAddress, power int64) {
+	// get validator signing info
+	signInfo, _ := k.slashingKeeper.GetValidatorSigningInfo(ctx, consAddr)
+
+	// do nothing if validator is jailed
+	if ctx.BlockTime().UnixNano() < int64(signInfo.JailedUntil.UnixNano()) {
+		return
+	}
+
+	// increase jail time
+	signInfo.JailedUntil = ctx.BlockHeader().Time.Add(k.slashingKeeper.DowntimeJailDuration(ctx))
+
+	// reset the missed block counters
+	signInfo.MissedBlocksCounter = 0
+	signInfo.IndexOffset = 0
+	k.slashingKeeper.ClearValidatorMissedBlockBitArray(ctx, consAddr)
+
+	// update signing info
+	k.slashingKeeper.SetValidatorSigningInfo(ctx, consAddr, signInfo)
```

Comment:
It’s a matter of opinion I guess. By adding the logic into the slashing module, it would introduce operations specific to consumer chains and not provider chains. I believe we prefer to have this kind of distinction implemented in the CCV module only.

Reply:
Yeah, I guess this approach make sense for V1 where we try to keep the changes to the rest of the SDK to the minimum. The alternative would be to have a global flag on the consumer chain indicating that the chain is a consumer chain, and add specific logic for that, but I agree, we don't need it for V1. 

BTW, in the slashing module we should make sure that this [line](https://github.com/cosmos/cosmos-sdk/blob/fa19ad5f85c540000e146c4c4f40631f68cc66f9/x/slashing/keeper/infractions.go#L78)  doesn't actually return a validator (the staking module on the consumer chain may have validators with the same keys as the ones coming from the provider). @jtremback this could be possible, right?